### PR TITLE
events: document code for the "Kill all enemies" check

### PR DESCRIPTION
**Pull request by kemenaran**

### CODE REVIEW

File: src/code/events.asm [hunk 1]
```diff
@@ -505,44 +505,57 @@ CheckKillInOrderTrigger::
...
+    ; resolved.
+
+    ; If hMultiPurpose0 != 08, then the trigger is resolved.
     ldh  a, [hMultiPurpose0]                      ; $6017: $F0 $D7
     cp   $08                                      ; $6019: $FE $08
-    jr   nz, .jr_6029                             ; $601B: $20 $0C
+    jr   nz, .triggerSatisfied                    ; $601B: $20 $0C
 
+    ; If wD460 == 0, don't mark the trigger as resolved.
     ld   a, [wD460]                               ; $601D: $FA $60 $D4
```
Comment (kemenaran) [inline src/code/events.asm:546]: These conditions are probably for the "Kill enemies in the right order" trigger in D2.

Comment (tobiasvl) [reply src/code/events.asm:546]: Maybe, but that's a separate trigger, though?

File: src/code/events.asm [hunk 2]
```diff
@@ -505,44 +505,57 @@ CheckKillInOrderTrigger::
...
-    cp   $FF                                      ; $6013: $FE $FF
-    jr   nz, jr_002_6000                          ; $6015: $20 $E9
+    cp   -1                                       ; $6013: $FE $FF
+    jr   nz, .loopOverEntities                    ; $6015: $20 $E9
 
+    ; We enumerated all entities, and all of them are either disabled
+    ; or excluded from the "Kill all" trigger: the trigger might be
+    ; resolved.
+
+    ; If hMultiPurpose0 != 08, then the trigger is resolved.
```
Comment (tobiasvl) [inline src/code/events.asm:540]: I wonder what this is?

**Review by tobiasvl — APPROVED** [review]

File: src/code/events.asm (hunk 1 above)
Comment (daid) [reply src/code/events.asm:546]: It's not the order. It's only referenced from `EntityFallHandler`. Which I assume handles an entity falling in a pit. I assume it's a "trigger" for the event, so it only happens when something is killed, and not all the time.

File: src/code/events.asm (hunk 2 above)
Comment (kemenaran) [reply src/code/events.asm]: I checked, and the caller sets it to the trigger type. I added this to the PR.

Comment (daid) [reply src/code/events.asm]: `hMultiPurpose0` stores the event trigger in this code (see 5FA1). So this checks for `TRIGGER_KILL_SPECIALS`

File: src/code/events.asm (hunk 1 above)
Comment (kemenaran) [reply src/code/events.asm:546]: Yup, and seems to be set only when the falling enemy is a `ENTITY_OPT1_EXCLUDED_FROM_KILL_ALL`. Hmm.

Comment (daid) [reply src/code/events.asm:546]: Seeing the other comment, I guess it's related to the special kill trigger?

**Pull request merged**